packages/babel-plugin-minify-dead-code-elimination/src/index.js

```diff
@@ -238,10 +247,25 @@ module.exports = ({ types: t, traverse }) => {
             }
 
             if (binding.references === 1 && binding.kind !== "param" && binding.kind !== "module" && binding.constant) {
+
               let replacement = binding.path.node;
               let replacementPath = binding.path;
+              let isAfterReturn = false;
+
               if (t.isVariableDeclarator(replacement)) {
-                replacement = replacement.init;
+
+                forEachPrevSibling(replacementPath, (path) => {
+                  if (t.isReturnStatement(path)) {
```

Comment:
Case 1 can be solved without a CFG ... `siblingPath.isAncestor(referencedPath)` will tell you this.

```js
function foo() {
  (() => {
    return x // x - ReferencedPath
  })() // CallExpression SiblingPath
  var x = 1;
}
```